src/TreeModelVolumes.cpp

```diff
@@ -148,9 +144,8 @@ const Polygons& TreeModelVolumes::calculateInternalModel(const RadiusLayerPair&
 {
     const auto& radius = key.first;
     const auto& layer_idx = key.second;
-    const auto& internal_areas = getAvoidance(radius, layer_idx).difference(getCollision(radius, layer_idx));
 
-    const std::lock_guard<std::mutex> lock(object_mutex_);
+    const auto& internal_areas = getAvoidance(radius, layer_idx).difference(getCollision(radius, layer_idx));
```

Comment:
auto.... vroem vroem! This makes things so much more readable imo. Keep using that